src/Symfony/Bundle/FrameworkBundle/Test/WebTestAssertions.php

```diff
@@ -0,0 +1,382 @@
+    {
+        $inputNode = static::getCrawler()->filter("input[name=\"$fieldName\"]");
+
+        Assert::assertGreaterThan(0, $inputNode->count(), "Input with name [$fieldName] not found on the page.");
+
+        Assert::assertEquals(
+            $expectedValue,
+            $value = $inputNode->getNode(0)->getAttribute('value'),
+            "Expected value [$value] for the [$fieldName] input does not equal the actual value [$value]."
+        );
+    }
+
+    public static function assertInputValueNotEquals(string $fieldName, string $expectedValue): void
+    {
+        $inputNode = static::getCrawler()->filter("input[name=\"$fieldName\"]");
+
+        Assert::assertGreaterThan(0, $inputNode->count(), "Input with name [$fieldName] not found on the page.");
+
+        Assert::assertNotEquals(
+            $expectedValue,
+            $value = $inputNode->getNode(0)->getAttribute('value'),
+            "Expected value [$value] for the [$fieldName] input was expected to not equal the actual value [$value].");
+    }
+
+    public static function assertRouteEquals($expectedRoute, array $parameters = []): void
+    {
+        $request = static::checkRequestAvailable();
+
+        Assert::assertSame(
+            $expectedRoute,
+            $route = $request->attributes->get('_route'),
+            "Expected route name [$expectedRoute] does not match the actual value [$route]."
+        );
+
+        if (\count($parameters)) {
+            foreach ($parameters as $key => $expectedValue) {
+                static::assertRequestAttributeValueEquals($key, $expectedValue);
+            }
+        }
+    }
+
+    public static function assertRequestAttributeValueEquals(string $key, $expectedValue): void
+    {
+        $request = static::checkRequestAvailable();
+
+        Assert::assertSame(
+            $expectedValue,
+            $value = $request->attributes->get($key),
+            "Expected request attribute [$key] value [$expectedValue] does not match actual value [$value]."
+        );
+    }
+
+    protected static function getResponseCookieFromClient(string $name, string $path = '/', string $domain = null): ?Cookie
+    {
+        $cookies = static::getResponse()->headers->getCookies();
+
+        $filteredCookies = \array_filter($cookies, function (Cookie $cookie) use ($name, $path, $domain) {
+            return
+                $cookie->getName() === $name
+                && $cookie->getPath() === $path
+                && $cookie->getDomain() === $domain
+            ;
+        });
+
+        return \reset($filteredCookies) ?: null;
+    }
+
+    private static function getClientForAssertion(): Client
+    {
+        if (!static::$client instanceof Client) {
+            static::fail(\sprintf(
+                'A client must be set to make assertions on it. Did you forget to call "%s::%s"?',
+                static::class, 'createClient'
```

Comment:
createClient can be part of the string instead of an argument to `sprintf`

Reply:
Thanks for the catch, fixed